src/conn/conn_api.c

```diff
@@ -1102,18 +1102,23 @@ __conn_is_new(WT_CONNECTION *wt_conn)
 static int
 __conn_close(WT_CONNECTION *wt_conn, const char *config)
 {
+    struct timespec cur_time, timer_start;
```

Comment:
It probably makes me a pedant, but the naming here doesn't seem consistent.

Could it be `timer_cur, timer_start` instead?

The pattern is common enough, I wonder if we should try to make it more natural? Adding something like a:

```
int __wt_timer_start(WT_TIMER *timer);
int __wt_timer_evaluate(WT_TIMER *timer, uint64_t *time_diff);
```
Used like:
```
WT_TIMER timer;
uint64_t time_diff;

__wt_timer_start(&timer);
/* Do things */
__wt_timer_evaluate(&timer, &time_diff);
/* Do what you want with the elapsed time. */
```

I know it's only the difference between a local variable and one line of code, but I think I'd find it more natural.

Reply:
Thanks for the suggestion. I changed the code to the new methods. I left most of the existing use cases, I feel it would be better to handle them in a different ticket.